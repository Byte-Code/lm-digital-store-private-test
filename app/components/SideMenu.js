import React, { Component } from 'react';
import styled from 'styled-components';
import Drawer from 'material-ui/Drawer';

import FlowerIcon from 'material-ui/svg-icons/maps/local-florist';
import FamilySideBar from '../containers/FamilySideBar';

const DivWithProps = (props) => <div {...props} />;

const Button = styled(DivWithProps)`
  width: 90px
  height: 120px;
  background: #339900;
  position: fixed;
  left: ${props => (props.pLeft || 0)};
  top: 70px;
  z-index: 2;
  cursor: pointer;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  box-shadow: rgba(0, 0, 0, 0.156863) 0px 3px 10px, rgba(0, 0, 0, 0.227451) 0px 3px 10px;
  display: ${props => (props.isVisible ? 'flex' : 'none')};
  &>p {
    text-align: center;
    color: #fff;
    font-size: 16px;
    font-family: LeroyMerlinSans Italic;
  }
`;

const Column = styled.div`
  background: ${props => props.bgColor || '#fff'};
  width: ${props => props.width || 'auto'};
  box-shadow: ${props => props.bShadow || 'none'};
  overflow: auto;
`;

export default class SideMenu extends Component {
  static propTypes = {}

  constructor(props) {
    super(props);
    this.state = { open: false };
  }

  toggleMenu = () => {
    this.setState({ open: !this.state.open });
  }

  render() {
    const { open } = this.state;

    return (
      <div>
        <Button onTouchTap={this.toggleMenu} isVisible={!open}>
          <FlowerIcon color="#fff" style={{ height: 50, width: 50 }} />
          <p>Giardino & Terrazzo</p>
        </Button>
        <Drawer
          docked={false}
          width={365}
          open={open}
          onRequestChange={(value) => this.setState({ open: value })}
          containerStyle={{ background: 'transparent', boxShadow: 'none', display: 'flex' }}
        >
          <Column
            bgColor="#fff"
            width="275px"
            bShadow="rgba(0, 0, 0, 0.156863) 0px 3px 10px, rgba(0, 0, 0, 0.227451) 0px 3px 10px"
          >
            <FamilySideBar />
          </Column>
          <Column bgColor="transparent">
            <Button
              onTouchTap={this.toggleMenu}
              pLeft="275px"
              isVisible={open}
            >
              <FlowerIcon color="#fff" style={{ height: 50, width: 50 }} />
              <p>Giardino & Terrazzo</p>
            </Button>
          </Column>
        </Drawer>
      </div>
    );
  }
}
