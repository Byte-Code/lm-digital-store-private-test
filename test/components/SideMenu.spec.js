import React from 'react';
import { shallow } from 'enzyme';

import SideMenu, { Button } from '../../app/components/SideMenu';

describe('SideMenu', () => {
  const result = shallow(
    <SideMenu />
  );

  it('should initially render closed', () => {
    expect(result.state('open')).toBe(false);
    expect(result).toMatchSnapshot();
  });

  it('the first button should be initially visible', () => {
    expect(result.find(Button).at(0).prop('isVisible')).toBe(true);
  });

  it('should open once the first button is clicked', () => {
    result.find(Button).at(0).simulate('click');
    expect(result.state('open')).toBe(true);
    expect(result).toMatchSnapshot();
  });

  it('the first button should not be visible now', () => {
    expect(result.find(Button).at(0).prop('isVisible')).toBe(false);
  });

  it('should close again once the second button is clicked', () => {
    result.find(Button).at(1).simulate('click');
    expect(result.state('open')).toBe(false);
    expect(result).toMatchSnapshot();
  });
});
