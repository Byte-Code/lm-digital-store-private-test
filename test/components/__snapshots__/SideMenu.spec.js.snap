// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu should close again once the second button is clicked 1`] = `
<div>
  <styled.div
    isVisible={true}
    onClick={[Function]}
  >
    <MapsLocalFlorist
      color="#fff"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <p>
      Giardino & Terrazzo
    </p>
  </styled.div>
  <Drawer
    containerStyle={
      Object {
        "background": "transparent",
        "boxShadow": "none",
        "display": "flex",
      }
    }
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={false}
    openSecondary={false}
    swipeAreaWidth={45}
    width={365}
    zDepth={2}
  >
    <styled.div
      bShadow="rgba(0, 0, 0, 0.156863) 0px 3px 10px, rgba(0, 0, 0, 0.227451) 0px 3px 10px"
      bgColor="#fff"
      width="275px"
    >
      <Connect(FamilySideBar)
        closeMenu={[Function]}
      />
    </styled.div>
    <styled.div
      bgColor="transparent"
    >
      <styled.div
        isVisible={false}
        onClick={[Function]}
        pLeft="275px"
      >
        <MapsLocalFlorist
          color="#fff"
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
        <p>
          Giardino & Terrazzo
        </p>
      </styled.div>
    </styled.div>
  </Drawer>
</div>
`;

exports[`SideMenu should initially render closed 1`] = `
<div>
  <styled.div
    isVisible={true}
    onClick={[Function]}
  >
    <MapsLocalFlorist
      color="#fff"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <p>
      Giardino & Terrazzo
    </p>
  </styled.div>
  <Drawer
    containerStyle={
      Object {
        "background": "transparent",
        "boxShadow": "none",
        "display": "flex",
      }
    }
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={false}
    openSecondary={false}
    swipeAreaWidth={45}
    width={365}
    zDepth={2}
  >
    <styled.div
      bShadow="rgba(0, 0, 0, 0.156863) 0px 3px 10px, rgba(0, 0, 0, 0.227451) 0px 3px 10px"
      bgColor="#fff"
      width="275px"
    >
      <Connect(FamilySideBar)
        closeMenu={[Function]}
      />
    </styled.div>
    <styled.div
      bgColor="transparent"
    >
      <styled.div
        isVisible={false}
        onClick={[Function]}
        pLeft="275px"
      >
        <MapsLocalFlorist
          color="#fff"
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
        <p>
          Giardino & Terrazzo
        </p>
      </styled.div>
    </styled.div>
  </Drawer>
</div>
`;

exports[`SideMenu should open once the first button is clicked 1`] = `
<div>
  <styled.div
    isVisible={false}
    onClick={[Function]}
  >
    <MapsLocalFlorist
      color="#fff"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <p>
      Giardino & Terrazzo
    </p>
  </styled.div>
  <Drawer
    containerStyle={
      Object {
        "background": "transparent",
        "boxShadow": "none",
        "display": "flex",
      }
    }
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={true}
    openSecondary={false}
    swipeAreaWidth={45}
    width={365}
    zDepth={2}
  >
    <styled.div
      bShadow="rgba(0, 0, 0, 0.156863) 0px 3px 10px, rgba(0, 0, 0, 0.227451) 0px 3px 10px"
      bgColor="#fff"
      width="275px"
    >
      <Connect(FamilySideBar)
        closeMenu={[Function]}
      />
    </styled.div>
    <styled.div
      bgColor="transparent"
    >
      <styled.div
        isVisible={true}
        onClick={[Function]}
        pLeft="275px"
      >
        <MapsLocalFlorist
          color="#fff"
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          }
        />
        <p>
          Giardino & Terrazzo
        </p>
      </styled.div>
    </styled.div>
  </Drawer>
</div>
`;
